### DISCUSSION TITLE
min and max value for LinearProgress?

### DISCUSSION THREAD

**Discussion by atticus-lv** [zauberzeug/nicegui#2937] [Ideas / Feature Requests]
Hi, I was wonder that why the LinearProgress do not have the same min max value as CircularProgress.
It shows 0 to 1 instead of 1-100(or 1~100%). In my opinion, it is not so friendly for the viewer (but still acceptable)
Maybe a separate label/progress is good for both of them?

**Comment by falkoschindler** [comment]
Hi @atticus-lv,

With `ui.linear_progress` based on Quasar's [QLinearProgress](https://quasar.dev/vue-components/linear-progress) and `ui.circular_progress` based on [QCircularProgress](https://quasar.dev/vue-components/circular-progress), we decided to stick to their API. Strangely though, QLinearProgress doesn't come with `min` and `max` props and we didn't implement them ourselves yet. This might be a useful API extension.

Providing a format string for the current value would also be a nice addition. Using [formatted string literals](https://docs.python.org/3/tutorial/inputoutput.html#formatted-string-literals) would allow the user to choose between 0.0..1.0 and 0%..100%.

**Comment by atticus-lv** [comment]
I got it,thanks~

**Reply by falkoschindler** [reply]
Let's keep this feature request open. Maybe someone from the community would like to implement it. 🙂

**Comment by jeffective** [comment]
it is possible to customize / decouple the value readout from the underlying value. After examining the source code for ui.linear_progress, I came up with the following solution

```python
def draw_sequence_linear_progress(backend: AutosequenceBackend):
    def format_main_seq_progress(progress: float | None) -> str:
        if progress is None:
            return ""
        return f"{progress:.1%}" if progress != 1 else "Main Sequence Completed"

    def format_abort_seq_progress(progress: float | None) -> str:
        if progress is None:
            return ""
        return f"{progress:.1%}" if progress != 1 else "Abort Completed"

    with (
        ui.linear_progress(color="green", show_value=False, size="20px")
        .props("instant-feedback")
        .bind_value_from(backend, "sequence_frac_complete")
        .bind_visibility_from(backend, "show_main_seq_progress")
    ) as progress:
        ui.label().classes("absolute-center text-sm text-white").bind_text_from(
            progress, "value", backward=format_main_seq_progress
        )
    with (
        ui.linear_progress(color="red", show_value=False, size="20px")
        .props("instant-feedback")
        .bind_value_from(backend, "sequence_frac_complete")
        .bind_visibility_from(backend, "show_abort_seq_progress")
    ) as progress:
        ui.label().classes("absolute-center text-sm text-white").bind_text_from(
            progress, "value", backward=format_abort_seq_progress
        )
```

which looks like this:
![image](https://github.com/zauberzeug/nicegui/assets/56497124/fb859fcc-6716-4248-8355-1b66c625169b)

![image](https://github.com/zauberzeug/nicegui/assets/56497124/e3d52366-941c-44c7-a464-a133ab642420)


(for a machine control application)
